### ISSUE TITLE
Not working as expression output_field in annotation/aggregation

### ISSUE THREAD

**Issue by OJFord** [danni/django-postgres-composite-types#27]
```python
Model.objects.annotate(composite=Func(1, 2, 3, function="construct_composite", output_field=CompositeField()))
```

results in

```python
model.composite == '(1,2,3)'
```

i.e. a `str` rather than the expected (hoped for) `CompositeType`.

**Comment by danni** [comment]
Keen for someone to have a go at this. A PR with a failing test case would also be a great help.

**Comment by OJFord** [comment]
I do intend to try to get a test case for it - because weirdly I've actually seen it both ways now.

In our tests, and hence I initially thought it not working, it came through as a string as described. Then we saw an issue with it actually running where it was the composite type (the issue being that we tried to change it into one ourselves assuming a string, and it wasn't a string, so some string function didn't exist).

It's not immediately apparent whether the test is doing something differently and both behaviours could be seen in prod too (according to how the QuerySet is constructed, or something) or if it's some how inherent in test vs. real (quirk of its interaction with pytest-django, say). Not got to the bottom of it yet.
